plugins/user-settings-backend/src/service/router.ts

```diff
@@ -104,6 +108,14 @@ export async function createRouterInternal(options: {
       key,
     });
 
+    if (options.signals) {
+      await options.signals.publish<{ key: string }>({
```

Comment:
signals generally would deserve to have dedicated named types right? like

```ts
export interface SignalKeyChanged {
  type: 'key-changed',
  key: string;
}
```

exported from a -common package, right?

Reply:
Yeah, that requires a completely new plugin as there's not `user-settings-common` available yet.